### PR TITLE
Service butterfly decoupling

### PR DESCRIPTION
**Pull request by bodymindarts**
This PR removes the dependency from Service to butterfly::Server by adding the ServiceConfig and ServiceFile struct to the census module.

Part of https://github.com/habitat-sh/habitat/issues/2094.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/manager/mod.rs [hunk 1]
```diff
@@ -509,46 +512,40 @@ impl Manager {
...
-                // handle this and not potentially panic
-                match service.config.to_exported() {
-                    Ok(cfg) => {
-                        *rumor.mut_cfg() =
-                            toml::ser::to_vec(&cfg).expect("Can't serialize to TOML bytes")
-                    }
-                    Err(err) => warn!("Error loading service config after update, err={}", err),
-                }
-                self.butterfly.insert_service(rumor);
+                self.gossip_latest_service_rumor(&service);
```
Comment (reset) [inline components/sup/src/manager/mod.rs]: NICE refactor

File: components/sup/src/manager/service/mod.rs [hunk 2]
```diff
@@ -67,23 +66,24 @@ lazy_static! {
 
 #[derive(Debug, Serialize)]
 pub struct Service {
```
Comment (reset) [inline components/sup/src/manager/service/mod.rs]: I like this pattern of putting a new line between pub and private members

File: components/sup/src/manager/mod.rs [hunk 3]
```diff
@@ -450,7 +452,8 @@ impl Manager {
                     .write()
                     .expect("Services lock is poisoned!")
                     .iter_mut() {
-                if service.tick(&self.butterfly, &self.census_ring) {
+                if service.tick(&self.census_ring) {
```
Comment (reset) [inline components/sup/src/manager/mod.rs]: This is much, much cleaner

**Comment by reset** [comment]
*excellent* work @bodymindarts!

tenor-219221377

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/226138367).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/226138367).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**